### PR TITLE
Consumer groups loading performance impr

### PR DESCRIPTION
**Pull request by iliax**
1. ACL applied on entire group list
2. loading all group descriptions before paginating only for MEMBERS sorting

- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [x] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ConsumerGroupService.java [hunk 1]
```diff
@@ -97,81 +98,92 @@ private boolean isConsumerGroupRelatesToTopic(String topic,
...
+        var comparator = Comparator.<ConsumerGroupDescription>comparingInt(cg -> cg.members().size());
+        var groupNames = groups.stream().map(ConsumerGroupListing::groupId).toList();
+        yield ac.describeConsumerGroups(groupNames)
+            .map(descriptions ->
+                descriptions.values()
+                    .stream()
+                    .sorted(sortOrderDto == SortOrderDTO.ASC ? comparator : comparator.reversed())
+                    .skip((long) (pageNum - 1) * perPage)
+                    .limit(perPage)
+                    .toList()
```
Comment (iliax) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ConsumerGroupService.java:163]: try to generalize

**Review by Haarolean — APPROVED** [review]

**Comment by yardenshoham** [comment]
How can one add a sort by messages behind? As opposed to NAME, STATE, and MEMBERS - we don't have the messages behind in the `loadSortedDescriptions` function

**Comment by Haarolean** [comment]
> How can one add a sort by messages behind? As opposed to NAME, STATE, and MEMBERS - we don't have the messages behind in the `loadSortedDescriptions` function

#3468

**Comment by yardenshoham** [comment]
Yes, I opened it. Just trying to see how it can be done

**Pull request merged**